### PR TITLE
Add read the docs

### PR DESCRIPTION
**Pull request by danlamanna**
@mgrauer Let me know what you think.

I have a few nitpicks, but I think this is a decent start.

### CODE REVIEW

File: docs/creating-a-source.rst [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+Add the widget constructed and rendered by the event handler
+
+::
+
+   web_external/js/views/widgets/ElasticsearchWidget.js
+   web_external/templates/widgets/elasticsearchWidget.jade
+
+Comply with Javascript styles
+-----------------------------
+Because it's the law of the land.
```
Comment (jbeezley) [inline docs/creating-a-source.rst]: :cop:

**Comment by mgrauer** [comment]
LGTM, but Travis failed.  Something about ssl verify in requests.  My first approach of course is to not think about it, restart Travis, and hope the problem goes away.

**Comment by aashish24** [comment]
Nice!! @danlamanna @mgrauer

**Comment by mgrauer** [comment]
We are getting some 2 test failures.  I doubt they are related to this work, but haven't had time to investigate.  One is with the contour analysis test and one is with the mean contour analysis test.

For MCA, we are getting 

>127: ======================================================================
127: ERROR: testAnalysis (plugin_tests.mean_contour_analysis_test.MeanContourAnalysisTestCase)
127: ----------------------------------------------------------------------
127: Traceback (most recent call last):
127:   File "/home/travis/build/girder/plugins/minerva/plugin_tests/mean_contour_analysis_test.py", line 247, in testAnalysis
127:     outputs = romanesco.run(romAnalysis, data['inputs'], data['outputs'])
127:   File "/home/travis/virtualenv/python2.7.9/src/romanesco/romanesco/utils.py", line 285, in wrapped
127:     return fn(*args, **kwargs)
127:   File "/home/travis/virtualenv/python2.7.9/src/romanesco/romanesco/__init__.py", line 268, in run
127:     auto_convert=auto_convert, validate=validate, **kwargs)
127:   File "/home/travis/virtualenv/python2.7.9/src/romanesco/romanesco/plugins/spark/pyspark_executor.py", line 31, in run
127:     exec task["script"] in custom.__dict__
127:   File "<string>", line 231, in <module>
127:   File "/home/travis/build/girder/clients/python/girder_client/__init__.py", line 613, in downloadFile
127:     headers={'Girder-Token': self.token})
127:   File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/requests/api.py", line 69, in get
127:     return request('get', url, params=params, **kwargs)
127:   File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/requests/api.py", line 50, in request
127:     response = session.request(method=method, url=url, **kwargs)
127:   File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/requests/sessions.py", line 468, in request
127:     resp = self.send(prep, **send_kwargs)
127:   File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/requests/sessions.py", line 597, in send
127:     history = [resp for resp in gen] if allow_redirects else []
127:   File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/requests/sessions.py", line 195, in resolve_redirects
127:     **adapter_kwargs
127:   File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/requests/sessions.py", line 576, in send
127:     r = adapter.send(request, **kwargs)
127:   File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/requests/adapters.py", line 433, in send
127:     raise SSLError(e, request=request)
127: Exception: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:581)

For the contour analysis test, the error we are getting is that `output_dir` is not defined.  I suspect that there is some other exception (probably the same SSL type error most likely [here](https://github.com/Kitware/minerva/blob/90cc0d050efe75bef85b16a763e05f390847abd4/analyses/NEX/geo_js_contour/contour.py#L75-L76)) getting thrown, and when the [finally](https://github.com/Kitware/minerva/blob/master/analyses/NEX/geo_js_contour/contour.py#L99-L103) block runs, `output_dir` hasn't been defined.

Travis is also not showing the status of the builds in the PR, so it is helpful to go to

https://travis-ci.org/Kitware/minerva/builds

**Comment by danlamanna** [comment]
I've started looking into this on a separate branch - something about this PR and my force push destroyed Travis for it. Will update once it's finished.

**Comment by jbeezley** [comment]
What I'm curious about is why there is an ssl error there at all.  That request to the local girder instance shouldn't even be using ssl right?

**Comment by danlamanna** [comment]
We're having a number of issues with Travis right now - but the changes from this branch were placed in #153 which built and merged successfully.

**Pull request closed without merging**